src/main/java/net/fabricmc/loom/task/AbstractRunTask.java

```diff
@@ -135,7 +151,49 @@ public void exec() {
 		setWorkingDir(new File(getProjectDir().get(), getInternalRunDir().get()));
 		environment(getInternalEnvironmentVars().get());
 
-		super.exec();
+		// Wrap with XVFB if enabled and on Linux
+		if (shouldUseXvfb()) {
+			LOGGER.info("Using XVFB for headless client execution");
+			execWithXvfb();
+		} else {
+			super.exec();
+		}
+	}
+
+	private boolean shouldUseXvfb() {
+		return getUseXvfb().get();
+	}
+
+	private void execWithXvfb() {
+		String xvfbRunPath = "/usr/bin/xvfb-run";
+
+		// Get the java executable path - try toolchain first, fallback to system property
+		String javaExec;
+
+		if (getJavaLauncher().isPresent()) {
+			javaExec = getJavaLauncher().get().getExecutablePath().getAsFile().getAbsolutePath();
+		} else {
+			// Fallback: construct path from JAVA_HOME
+			String javaHome = System.getProperty("java.home");
+			javaExec = javaHome + File.separator + "bin" + File.separator + "java";
+		}
+
+		// Build the complete command line: xvfb-run --auto-servernum java [jvm-args] mainclass [program-args]
+		List<String> commandLine = new ArrayList<>();
+		commandLine.add(xvfbRunPath);
+		commandLine.add("--auto-servernum");
```

Comment:
Again, what do this do? We didnt set it before? or is it the same as `-a`?

Reply:
Same as -a, this is just more concise